projects/packages/search/src/dashboard/components/record-meter/barchart.jsx

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+const Chart = React.lazy( () => import( 'chart.js/auto' ) );
+
+/**
+ * Internal dependencies
+ */
+
+export class BarChart extends React.Component {
+	constructor( props ) {
+		super( props );
+		this.canvasRef = React.createRef();
+	}
+
+	componentDidMount() {
+		this.myChart = new Chart( this.canvasRef.current, {
+			type: 'bar',
+			options: {
+				borderRadius: 100,
+				borderSkipped: 'middle',
+				indexAxis: 'y',
+				maintainAspectRatio: false,
+				aspectRatio: 1.3,
+				scales: {
+					x: {
+						stacked: true,
+						grid: {
+							display: false,
+							drawBorder: false,
+						},
+						ticks: {
+							display: false,
+						},
+					},
+					y: {
+						stacked: true,
+						grid: {
+							display: false,
+							drawBorder: false,
+						},
+						ticks: {
+							display: false,
+						},
+					},
+				},
+
+				plugins: {
+					title: {
+						display: false,
+					},
+					subtitle: {
+						display: false,
+					},
+					legend: {
+						display: false,
+
+						labels: {
+							filter: function ( legendItem, data ) {
+								return ! legendItem.text.includes( 'Remaining' );
+							},
+						},
+					},
+				},
+			},
+
+			data: {
+				labels: [ '' ], // this empty label must remain, if removed the entire chart breaks
+				datasets: this.props.data.map( d => d.data ),
+			},
+		} );
+
+		if ( this.props.isValid === true ) {
+			this.setState( { legendItems: this.myChart.legend.legendItems } );
+		}
+	}
+
+	render() {
+		if ( this.props.isValid === false ) {
+			return null;
+		}
+		return (
+			<div className="barChart">
```

Comment:
Our CSS class convention here is `bar-chart`.

Reply:
And a prefix might better, to avoid CSS conflicts, for example `jps_dashboard_bar_chart`

Reply:
Good call. Should be dashes to separate the prefix though :)

The main dashboard uses `jp-search-dashboard-page`.

Maybe we could use `jp-search-record-meter-bar-chart`?